hubgrep/models/__init__.py

```diff
@@ -42,8 +47,27 @@ class HostingService(db.Model):
     # api keys for a backend?
     api_url = db.Column(db.String(500), unique=True, nullable=False)
 
+    # individual config for a specific service (eg. api-key)
     # could be json, but thats only supported for postgres
     config = db.Column(db.Text)
 
     # frontend label
     label = db.Column(db.String(80))
+
+    def set_service_label(self):
+        self.label = re.split("//", self.api_url)[1].rstrip("/")
+
+    def get_hosting_service_interface(self, cached_session: 'CachedSession', timeout: int):
+        config_str = self.config
+        config = json.loads(config_str)
+
+        HostingInterfaceClass = hosting_service_interface_mapping[self.type]
+        hosting_service_interface = HostingInterfaceClass(
```

Comment:
all hosting services have the exact same constructor now (before the GitLab class got the api_key, and we used some ugly unfolding of the config to do that. now all services get a config dict, even if its empty.)